### PR TITLE
LIRC Improvements

### PR DESCRIPTION
**Pull request by leon-anavi**
Move lirc to dynamic-layers/meta-python/recipes-connectivity/ to
avoid related issues while building rpi-basic-image.

Reported-by: Alan <engineer@iotexpert.com>
Suggested-by:  Khem Raj <raj.khem@gmail.com>
Signed-off-by: Leon Anavi <leon.anavi@konsulko.com>

**- What I did**

This GitHub pull request brings the following improvements for infrared support and LIRC:

- Fix building rpi-basic-image and other images
- Add variable `ENABLE_IR` to enable infrared support
- Include `gpio-ir` and `gpio-ir-tx` modules if `ENABLE_IR` is enabled
- Use the same `lircd.service` as in Raspbian wto start LIRC with argument `--nodaemon` after `lircd-setup.service`
- Add details about `ENABLE_IR` to the documentation

**- How I did it**

- Moved lirc bbappend file and patch to an appropriate location as discussed with Alan and @kraj in private emails. In the mean time I am spinning a build to verify the change. Thank you both for reporting and suggesting the fix for this issue. Sorry for the inconvenience caused by my previous commit.
- Added `ENABLE_IR` to rpi-config
- Included appropriate modules in `rpi-base.inc` if `ENABLE_IR` is set to 1 (similar approach to what I did recently for I2C)
- Update `lircd.service` in `lirc_%.bbappend` to use the same approach for launching LIRC as in Raspbian
- Added instructions to `docs/extra-build-config.md`

Thanks,
Leon

### CODE REVIEW

**Comment by leon-anavi** [comment]
Hi @kraj, @agherzan,

I've decided to extend to scope of this GitHub pull request with several improvements for the infrared support. Now with `ENABLE_IR = "1"` and `IMAGE_INSTALL_append = " lirc"` in `conf/local.conf` LIRC will work out of the box on the Raspberry Pi.

Best regards, Leon

File: dynamic-layers/meta-python/recipes-connectivity/lirc/lirc_%.bbappend [hunk 1]
```diff
@@ -2,4 +2,5 @@ FILESEXTRAPATHS_prepend := "${THISDIR}/${PN}:"
 
 SRC_URI_append_rpi = " \
 	file://lirc-gpio-ir-0.10.patch \
+        file://lircd.service \
```
Comment (agherzan) [inline dynamic-layers/meta-python/recipes-connectivity/lirc/lirc_%.bbappend:5]: NIT: alignment

File: docs/extra-build-config.md [hunk 2]
```diff
@@ -261,6 +261,18 @@ In order to use Pican2 CAN module, set the following variables:
...
 See: <http://skpang.co.uk/catalog/pican2-canbus-board-for-raspberry-pi-23-p-1475.html>
 
+## Enable infrared
+
+Users who want to enable infrared support, for example for using LIRC (Linux
+Infrared Remote Control), have to explicitly set in local.conf:
+
+    ENABLE_IR = "1"
+
+This will add device tree overlays gpio-ir and gpio-ir-tx will to config.txt.
```
Comment (agherzan) [inline docs/extra-build-config.md:271]: Small typo: "will".

Comment (leon-anavi) [reply docs/extra-build-config.md]: Thank you. Fixed. I've forced pushed a new version for review. Thanks, Leon

**Comment by leon-anavi** [comment]
> Raised some NITs but otherwise looks good.

Thank you for the quick feedback. I've fixed the typo and submitted a new version for a review.

Best regards, Leon

**Pull request merged**